File: BackEnd/src/main/java/com/restaurant/backend/Entity/Booking.java
package com.restaurant.backend.Entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;

@Entity
@Table(name = "bookings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // User = Customer
    @ManyToOne
    @JoinColumn(name = "customer_id", nullable = false)
    private User customer;

    @ManyToOne
    @JoinColumn(name = "table_id")
    private RestaurantTable table;

    @Column(name = "booking_date", nullable = false)
    private LocalDate date;

    @Column(name = "booking_time", nullable = false)
    private LocalTime time;

    private int guests;

    @Column(columnDefinition = "TEXT")
    private String note;

    private String status;

    @Column(name = "booking_code", unique = true, length = 20)
    private String bookingCode;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

}

File: mobileapp/app/src/main/java/com/restaurant/mobileapp/data/model/Booking.kt
package com.restaurant.mobileapp.data.model

import com.google.gson.annotations.SerializedName

data class Booking(
    val id: Long? = null,
    val date: String,
    val time: String,
    val guests: Int,
    val note: String? = null,
    val status: String = "PENDING",
    val customerCode: String? = null,
    val bookingCode: String? = null,
    val customer: User? = null,
    val table: RestaurantTable? = null
)

data class BookingRequest(
    val customerId: Long,
    val date: String,
    val time: String,
    val guests: Int,
    val note: String? = null,
    val tableId: Long? = null
)

data class BookingResponse(
    val message: String,
    val booking: Booking,
    val bookingCode: String? = null
)

data class BookingListResponse(
    val message: String,
    val bookings: List<Booking>
)
